src/named_tuple.cr

```diff
@@ -38,13 +38,16 @@ struct NamedTuple
   end
 
   # Creates a named tuple from the given hash, with elements casted to the given types.
+  # Here the Int32 | String union is cast to Int32.
   #
   # ```
-  # NamedTuple(foo: String, bar: Int64).from({:foo => "world", :bar => 2})       # => {foo: "world", bar: 2}
-  # NamedTuple(foo: String, bar: Int64).from({"foo" => "world", "bar" => 2})     # => {foo: "world", bar: 2}
-  # NamedTuple(foo: String, bar: Int64).from({:foo => "world", :bar => 2}).class # => {foo: String, bar: Int64}
-  # ```
+  # num_or_str : Int32 | String
+  # num_or_str = 42
```

Comment:
This does not creates a union:

```crystal
num_or_str : Int32 | String
num_or_str = 42
pp typeof(num_or_str) # => Int32
```

Either `num_or_str = 42 || "a"`, `num_or_str = 42.as(Int32 | String)` or `num_or_str = rand >= 0.0 ? 42 : "a"` are the usual idioms used for one liners unions.